tests/integrationv2/fixtures.py

```diff
@@ -94,8 +94,29 @@ def _fn(
         # is nothing to capture here.
         pass
     finally:
-        # Whether the processes succeeded or not, clean then up.
         for p in processes:
+            # Always print the results
+            if p.results:
```

Comment:
Okay so bear with me: ManagedProcess is async. We don't print stdout / stderr until after the results for a process are marked ready. Once the results for one process are marked ready, a test can assert the success of that process. If that assertion fails, we may exit the test without printing stdout / stderr. Usually this just results in failure to print stdout / stderr for one process, but because we mark the results ready BEFORE doing the print, theoretically we could fail to print stdout / stderr for either process.

Rather than trying to cleanup that async mess, I moved the printing here. This logic should ALWAYS trigger, regardless of failed assertions. Handling teardown logic is one of the specific responsibilities of fixtures.

Reply:
Code references to attempt to explain:
* get_results() waits for _results_ready(): [link](https://github.com/aws/s2n-tls/blob/main/tests/integrationv2/processes.py#L522-L534)
* _results_ready() waits for results to exist: [link](https://github.com/aws/s2n-tls/blob/d3ad42999373389b4dbd1a381a76024b54855248/tests/integrationv2/processes.py#L507-L509)
* results are set in run(): [link](https://github.com/aws/s2n-tls/blob/d3ad42999373389b4dbd1a381a76024b54855248/tests/integrationv2/processes.py#L448)
* stdout/stderr are printed later in run(): [link](https://github.com/aws/s2n-tls/blob/d3ad42999373389b4dbd1a381a76024b54855248/tests/integrationv2/processes.py#L478)